src/jmespath/TreeInterpreter.js

```diff
@@ -123,7 +123,7 @@ export default class TreeInterpreter {
 
       Index: (node, value) => {
         if (!isArray(value)) return null;
-        let index = node.value;
+        let index = this.toNumber(this.visit(node.value, value));
```

Comment:
`this.toNumber` will always return a number. Do we want to only coerce valid cases and throw an error for invalid cases here?

Reply:
We can consider having `toNumber()` be less forgiving -- but I think we'd want to be consistent and have it fail in all contexts -- not just here.  What do you think?

Reply:
I think I misunderstood the implementation. The current one looks alright since it forwards `null` values. Array and object coercions emit a debug event. 😄